test/suite/test_backup05.py

```diff
@@ -0,0 +1,113 @@
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_backup05.py
+#   Test that backups can be performed similar to MongoDB's fsyncLock.
+#   We assume writes are not being performed, a checkpoint is done and
+#   then we open a backup cursor to prevent log archiving and other file
+#   manipulations.  Manually copy the directory and verify it.
+#
+
+import fnmatch, os, shutil, subprocess, time
+from suite_subprocess import suite_subprocess
+from wiredtiger import wiredtiger_open
+from wtscenario import multiply_scenarios, number_scenarios, prune_scenarios
+import wttest
+
+class test_backup05(wttest.WiredTigerTestCase, suite_subprocess):
+    uri = 'table:test_backup05'
+    create_params = 'key_format=i,value_format=i'
+    freq = 5
+
+    def copy_windows(self, olddir, newdir):
+        os.mkdir(newdir)
+        for fname in os.listdir(olddir):
+            fullname = os.path.join(olddir, fname)
+            # Skip lock file on Windows since it is locked
+            if os.path.isfile(fullname) and "WiredTiger.lock" not in fullname:
+                shutil.copy(fullname, newdir)
+
+    def check_manual_backup(self, i, olddir, newdir):
+        ''' Simulate a manual backup from olddir and restart in newdir. '''
+        self.session.checkpoint()
+        cbkup = self.session.open_cursor('backup:', None, None)
+        # with the connection still open, copy files to new directory
+        shutil.rmtree(newdir, ignore_errors=True)
+
+        if os.name == "nt":
+            copy_windows(olddir, newdir)
+        else:
+            # Half the time use copytree, the other half use a dd command that
+            # does not align on a block boundary.
+            if i % (self.freq * 2) == 0:
+                os.mkdir(newdir)
+                for fname in os.listdir(olddir):
+                    fullname = os.path.join(olddir, fname)
+                    inpf = 'if=' + fullname
+                    outf = 'of=' + newdir + '/' + fullname
+                    cmd_list = ['dd', inpf, outf, 'bs=300']
+                    a = subprocess.Popen(cmd_list)
+                    a.wait()
+            else:
+                shutil.copytree(olddir, newdir)
+
+        # Now simulate fsyncUnlock by closing the backup cursor.
+        cbkup.close()
+
+        # Open the new directory and verify
+        conn = self.setUpConnectionOpen(newdir)
+        session = self.setUpSessionOpen(conn)
+        session.verify(self.uri)
+        conn.close()
+        
+    def test_backup(self):
+        '''Check manual fsyncLock backup strategy'''
+
+        # Here's the strategy:
+        #    - update the table
+        #    - checkpoint the database
+        #    - open a backup cursor
+        #    - copy the database directory (live, simulating a crash)
```

Comment:
@ddanderson, the principle here is that copying a live database then trying to open it is equivalent to what would happen if the process crashed at that point and had to recover.  There are additional things that can go wrong when the power fails that we aren't testing, but this catches a lot of problems.  We test recovery this way in several places.